### PR TITLE
feat: Allow registration of advanced external share actions

### LINKED ISSUE #43881: Add advanced external share actions
Subtask to #6841

### PR DESCRIPTION
**Pull request by Pytal**
- Resolves: https://github.com/nextcloud/server/issues/43881

## Summary

- Allow registering advanced external share actions that are now placed inside the advanced share settings
- Keep previous external share actions (https://github.com/search?type=code&q=org%3Anextcloud+%22OCA.Sharing.ExternalShareActions%22) rendered within the NcAction menu as this makes sense for UX
- Needed for the new One-time download entry according to mockup:
image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by Pytal** [comment]
We're using the non-public OCA API to keep the files_downloadlimit app separate as discussed :)

File: apps/files_sharing/src/views/SharingDetailsTab.vue [hunk 1]
```diff
@@ -860,6 +885,10 @@ export default {
 				this.queueUpdate(...permissionsAndAttributes)
 			}
 
+			await Promise.allSettled(this.$refs.externalLinkActions?.map((action) => {
+				return action.$children.at(0)?.onSave?.()
```
Comment (skjnldsv) [inline apps/files_sharing/src/views/SharingDetailsTab.vue:889]: If not defined in the action, we need to fallback to a `Promise.resolve()`, no?

**Review by skjnldsv — APPROVED** [review]

File: apps/files_sharing/src/views/SharingDetailsTab.vue [hunk 2]
```diff
@@ -318,6 +329,8 @@ export default {
 			isFirstComponentLoad: true,
 			test: false,
 			creating: false,
+
+			ExternalShareActions: OCA.Sharing.ExternalShareActions.state,
```
Comment (artonge) [inline apps/files_sharing/src/views/SharingDetailsTab.vue:333]: Why the capital E ?
```suggestion
			externalShareActions: OCA.Sharing.ExternalShareActions.state,
```

**Review by artonge — APPROVED** [review]

File: apps/files_sharing/src/views/SharingDetailsTab.vue (hunk 2 above)
Comment (Pytal) [reply apps/files_sharing/src/views/SharingDetailsTab.vue:333]: Copy and paste from https://github.com/nextcloud/server/blob/c18ffe0cad13dc5903cbb08ffbdb787f01db5e35/apps/files_sharing/src/components/SharingEntryLink.vue#L247 actually ;)

I'll keep it the same to be consistent

**Pull request merged**